hera_stats/jackknives.py

```diff
@@ -0,0 +1,402 @@
+    have the identifier and compares them to those which don't. It
+    automatically does every combination but only up to a maximum of
+    n_jacks (not yet tho).
+
+    If pairs are specified, which is a list of len-2 lists, the pairs are
+    selected using two indices provided. This is essentially a list of
+    pairs of indices to use for selecting files.
+
+    Parameters
+    ----------
+    uvp: list of UVPSpec objects
+        One entry to this list corresponds to one file, to be named in the
+        files kwarg.
+
+    files: list of strings
+        The filepaths of each corresponding UVPSpec in uvp.
+
+    identifier: string, optional
+        String segment to use as a filter for grouping files.
+
+    pairs: list, optional
+        List of index pairs, for a more manual selection of file groupings.
+
+    Returns
+    -------
+    uvpl: list
+        List of UVPSpecData objects split accordingly.
+    """
+    # Sanity checks
+    if type(uvp) != list:
+        raise AttributeError("Split files needs a list of uvp objects.")
+    if len(uvp) < 2:
+        raise AttributeError("Fewer than two files supplied. Make sure "
+                             "combine = False for load_uvd.")
+    if identifier is not None and filepairs is not None:
+        raise AttributeError("Please only specify either identifier or "
+                             "file pairs.")
+    if identifier is None and filepairs is None:
+        raise AttributeError("No identifier or file pair list specified.")
+
+    uvp = np.array(uvp)
+    files = np.array(files)
+    if type(identifier) == str:
+
+        infile = np.array([identifier in f for f in files])
+        grp1, uvp1 = files[infile], uvp[infile]
+        grp2, uvp2 = files[~infile], uvp[~infile]
+        minlen = min([len(uvp1), len(uvp2)])
+        for i in range(minlen):
+            uvp1[i].jkftype = "spl_files"
+            uvp1[i].labels = np.array([grp1[i]])
+            uvp2[i].jkftype = "spl_files"
+            uvp2[i].labels = np.array([grp2[i]])
+
+        if len(grp1) == 0:
+            raise AttributeError("Identifier not found in any filenames "
+                                 "loaded.")
+        if len(grp2) == 0:
+            raise AttributeError("Identifier found in all filenames... "
+                                 "Be more strict!")
+
+        print uvp1, uvp2
+        uvpl = [[uvp1[i], uvp2[i]] for i in range(len(uvp1))]
+        grps = [[grp1[i], grp2[i]] for i in range(len(grp1))]
+
+        if verbose:
+            print "Found %i pairs of files" % len(uvpl)
+
+    # Split according to index pairs provided
+    elif type(filepairs) == list:
+        uvpl = [[uvp[i], uvp[j]] for [i,j] in filepairs]
+        grps = [[files[i], files[j]] for [i,j] in filepairs]
+
+    return uvpl
+
+def split_times(uvp, periods=None, verbose=False):
+    """
+    Jackknife that splits the UVPSpecData into groups based on
+    the binsize.
```

Comment:
What is binsize?

Reply:
How does this deal with times that are completely flagged?

Reply:
(or maybe, more generally -- what if different time samples have very different nsamples?)